### PR TITLE
Update quick-start and changelog for VAULT_ASSUMED_ROLE_ARN

### PR DESCRIPTION
**Pull request by swenson**
Following up on https://github.com/hashicorp/vault-lambda-extension/pull/69

* Change `make build` target to build the ZIP file needed for the lambda
  extension layer.
* Added `TERRAFORM_ARGS` to `make quick-start` so that you can pass in
  variables.
* Updated quick start template to allow for running with the local
  lambda extension ZIP file.
* Updated quick start template to add an extra role to use with
  VAULT_ASSUMED_ROLE_ARN
* Updated the CHANGELOG
* Bumpbed Go version

I tested the following quick-start scenarios as working:

```sh
make quick-start
make quick-start TERRAFORM_ARGS="-var local_extension=true"
make quick-start TERRAFORM_ARGS="-var local_extension=true -var assume_role=true"
```

(I can't test `assume_role` without `local_extension` since we have not
made a release since the feature was added.)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,6 +10,8 @@ build: clean
 		-ldflags '-s -w' \
 		-a -o pkg/extensions/vault-lambda-extension \
 		.
+	cd pkg && zip -r vault-lambda-extension.zip extensions/
+	echo "Extension built: pkg/vault-lambda-extension.zip"
```
[the remark this answers is not in the document]
Comment (swenson) [reply Makefile:14]: Ah, good point. I'll move it to a separate target.

File: Makefile [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 
 build: clean
 	GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=0 go build \
 		-ldflags '-s -w' \
 		-a -o pkg/extensions/vault-lambda-extension \
 		.
 
+zip: build
+	cd pkg && zip -r vault-lambda-extension.zip extensions/
+	echo "Extension built: pkg/vault-lambda-extension.zip"
```
Comment (tvoran) [inline Makefile:16]:
```suggestion
	@echo "Extension built: pkg/vault-lambda-extension.zip"
```

File: README.md [hunk 3]
```diff
@@ -63,3 +63,12 @@ pricing.**
...
 [vault-aws-iam-auth]: https://www.vaultproject.io/docs/auth/aws
+
+## Testing
+
+If you want to test changes to the lambda extension, you can build and deploy the local version for testing with the Quick Start:
+
+```sh
+make zip
+make quick-start TERRAFORM_ARGS="-var local_extension=true"
+```
```
Comment (tvoran) [inline README.md:74]: Should we mention the assume_role option somewhere in the readme as well?

**Review by tvoran — APPROVED** [review]
Worked great for me!

File: README.md (hunk 3 above)
Comment (swenson) [reply README.md:74]: I have a separate PR in the Vault docs, which seemed like the right place to mention it https://github.com/hashicorp/vault/pull/16677

Comment (swenson) [reply README.md:74]: At least for the environment variable. I'll add a note to the README about the terraform variable.

**Comment by swenson** [comment]
Thanks!

File: README.md (hunk 3 above)
Comment (tvoran) [reply README.md:81]: Yep, sorry, the terraform variable is what I meant. 👍

**Pull request merged**